client/consensus/aura/src/import_queue.rs

```diff
@@ -176,8 +178,14 @@ where
 		if !inherent_res.ok() {
 			for (i, e) in inherent_res.into_errors() {
 				match create_inherent_data_providers.try_handle_error(&i, &e).await {
-					Some(res) => res.map_err(Error::Inherent)?,
+					// NOTE: we used to include an inherent with the slot number, which was used to
+					// validate the timestamp inherent. this inherent was deprecated and removed
+					// from the runtime but it is still around when validating historical blocks.
+					// in order to not keep around the slot inherent providers indefinitely we just
+					// ignore the unhandled inherent error.
+					None if i == sp_consensus_aura::AURA_DEPRECATED_INHERENT_IDENTIFIER => {},
```

Comment:
I suppose that in the past BABE/AURA pallets were implementing a check for "slot" inherent data and this has been removed a while ago...

I see that you've added the check to ignore this error. Can I ask under what conditions this case is supposed to happen? 

From what I understood this case may happen if you check a block without "slot" inherent data using an old runtime. 
If yes, assuming that a block without slot inherent can be executed at a lower chain height with an the old runtime, then:
- if such a block is executed using an old-client + an old-runtime then the check returns an error (because of the lack of BABE/AURA inherent data).
- But if I execute the same block with a new-client + old-runtime then the check returns ok, because the error is catched. But the result should be an error as well.

Is there something that I miss here? Forgive me if I wrote something stupid :-)